source/geod24/LocalRest.d

```diff
@@ -595,61 +558,51 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
-            }
-
             scheduler.start(() {
                 while (1)
                 {
-                    C.receiveTimeout(self, 10.msecs,
-                        (ShutdownCommand!API e)
-                        {
-                            if (e.callback !is null)
-                                e.callback(node);
-                            exc.restart = e.restart;
-                            throw exc;
-                        },
-                        (TimeCommand s) {
-                            control.sleep_until = Clock.currTime + s.dur;
-                            control.drop = s.drop;
-                        },
-                        (FilterAPI filter_api) {
-                            control.filter = filter_api;
-                        },
-                        (Response res) {
-                            if (!control.isSleeping())
-                                handle(res);
-                            else if (!control.drop)
-                                await_msgs ~= Variant(res);
-                        },
-                        (Command cmd)
+                    Variant msg;
+                    if (chn.read(msg, 10.msecs))
+                    {
+                        switch (msg.tag)
                         {
-                            if (!control.isSleeping())
-                                handle(cmd);
-                            else if (!control.drop)
-                                await_msgs ~= Variant(cmd);
-                        });
+                            case Variant.Type.command:
+                                Command cmd = msg.cmd;
+                                if (!control.isSleeping())
+                                    scheduler.spawn({
+                                        handleCommand(cmd, node, control.filter);
+                                    });
+                                else if (!control.drop)
+                                    await_msgs ~= msg;
+                                break;
+                            case Variant.Type.shutdown_command:
+                                ShutdownCommand!API e = msg.shutdown;
+                                if (e.callback !is null)
+                                    e.callback(node);
+                                exc.restart = e.restart;
+                                throw exc;
+                            case Variant.Type.time_command:
+                                TimeCommand s = msg.time;
+                                control.sleep_until = Clock.currTime + s.dur;
+                                control.drop = s.drop;
+                                break;
+                            case Variant.Type.filter:
+                                FilterAPI filter_api = msg.filter;
+                                control.filter = filter_api;
+                                break;
+                            default:
+                                assert(0);
```

Comment:
Note: `enum` are absolutely effing bs because you need to sanitize them EVERY. EFFING. TIME.
So even if your tag should only take values `[0 .. 4]` if correctly typed, you can actually receive `[0 .. 255]`.
`final switch` allows you to omit the `default` if you handle all values, but obviously that's just bad.
In this case, throwing an `Error` with the enum value would be good (there's a catch at the top level that will print an error message).

Reply:
Rule of thumb: Assume input to your function is provided by a vengeful ex with a PhD in cybersecurity.

Reply:
And here I thought she only cooked good pasta.